pkg/runtime/service_controller.go

```diff
@@ -124,6 +137,12 @@ func (c *serviceController) BindControllerManager(mgr ctrlrt.Manager, cfg ackcfg
 		}
 		c.reconcilers = append(c.reconcilers, rec)
 	}
+
+	rec := NewAdoptionReconciler(c, c.log, cfg, c.metrics)
+	if err := rec.BindControllerManager(mgr); err != nil {
+		return err
+	}
+	c.adoptionReconciler = rec
```

Comment:
nit/QQ: Would it be valuable to add a flag to enable/disable adoption reconciler? i'm thinking about `--enable-adoption-reconciler` and default it to `true`

Reply:
Since resources can be opted out of adoption (via the generator.yaml), I'm OK with adding a CLI flag after this first patch is merged. We can even consider a CRD that allows the service controller to be configured without restarting the service controller -- and that CRD could allow configuring things like the resources that should be adoptable.